src/main/webapp/app/exercises/programming/hestia/git-diff-report/git-diff-report.component.ts

```diff
@@ -108,12 +116,16 @@ export class GitDiffReportComponent implements OnInit {
         });
         this.leftCommit = this.report.leftCommitHash?.substring(0, 10);
         this.rightCommit = this.report.rightCommitHash?.substring(0, 10);
-        this.filePaths.forEach((path) => {
-            if (this.entriesByPath.get(path)?.length) {
-                this.diffsReadyByPath[path] = false;
-            }
-        });
-        this.nothingToDisplay = Object.keys(this.diffsReadyByPath).length === 0;
+        this.diffInformationForPaths = this.filePaths
+            .filter((path) => this.entriesByPath.get(path)?.length)
+            .map((path) => {
+                // entries is not undefined due to the filter
+                const entries = this.entriesByPath.get(path)!;
+                const templateFileContent = this.templateFileContentByPath.get(this.renamedFilePaths[path] || path);
+                const solutionFileContent = this.solutionFileContentByPath.get(path) ?? '';
```

Comment:
Is there a reason why you use `||` in one occasion but `??` in the other?

Reply:
The `??` was actually not even required here (`undefined` is a valid value for the solutionFileContent). I also changed the `renamedFilePaths` s.t. typescript now acknowledges that `renamedFilePaths[path]` can be undefined and replaced `||` with `??`.